packages/next-app/src/app/stableswapsimulator/(components)/ShareButton.tsx

```diff
@@ -0,0 +1,65 @@
+import { Share1Icon } from "@radix-ui/react-icons";
+import { usePathname } from "next/navigation";
+import { useState } from "react";
+
+import Button from "#/components/Button";
+import { Toast, ToastProvider } from "#/components/Toast";
+import { Tooltip } from "#/components/Tooltip";
+import { useStableSwap } from "#/contexts/StableSwapContext";
+
+export function ShareButton() {
+  const { generateURL } = useStableSwap();
+
+  const onInitialPage = usePathname() === "/stableswapsimulator";
+  const [isNotifierOpen, setIsNotifierOpen] = useState<boolean>(false);
+
+  return (
+    <ToastProvider>
```

Comment:
`<ToastProvider>` is not needed here as it is being invoked on the `<RootLayout>`